zkevm-circuits/src/evm_circuit/execution/return_revert.rs

```diff
@@ -94,11 +95,12 @@ impl<F: Field> ExecutionGadget<F> for ReturnRevertGadget<F> {
 
         let is_contract_deployment =
             is_create.clone() * is_success.expr() * not::expr(copy_rw_increase_is_zero.expr());
-        let (caller_id, address, reversion_info, code_hash) =
+        let (caller_id, address, reversion_info, code_hash, init_code_rlc) =
```

Comment:
agree with this . should be "deployed_code"

Reply:
for the `cb.keccak_table_lookup( 
             init_code_rlc.expr(), 
             init_code.length(), 
             create.code_hash_word_rlc(cb), 
         ); `, it is used to the computation of the newly created contract address. So that keccak_table_lookup needs init_code instead of deployment_code?

Use bytecode circuit to constraint this keccak hash:   as long as current code is correct i would prefer this.  For similar reasons with https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1316#issuecomment-1467265567. For example, for execution compatibility we use keccak for create/create2 contract address computation while in bytecode circuit we use poseidon.   So "bytecode circuit also do keccak" at least not works for us.    Since putting a keccak lookup here bring almost no performance cost compared to "let bytecode circuit do" method, so i think it is ok?

Reply:
> it is used to the computation of the newly created contract address. So that keccak_table_lookup needs init_code instead of deployment_code?

yes, what I wanted to ask is should we do the same thing for `RETURN`, but I just see it now https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/5e13f2b6f4ec919d8427d29db68de84a3c5bfabd/zkevm-circuits/src/evm_circuit/execution/return_revert.rs#L104-L115

We indeed copy the `deployed_code` to `bytecode_table` by this `copy_table` lookup.

One last thing I feel missing is we have to check this `code_hash` indeed has length `range.length()`, otherwise the prover could hide more bytes in the end, right?

> For example, for execution compatibility we use keccak for create/create2 contract address computation while in bytecode circuit we use poseidon. So "bytecode circuit also do keccak" at least not works for us.

I see, I didn't think about that, using another keccak for hash of `init_code` makes sense to me now.

Reply:
fixed `init_code` to `deploy_code_rlc` in [55aebf9](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1419/commits/55aebf9749918e6710173771d759636975d60d57)